internal/vds/connection.go

```diff
@@ -63,6 +67,46 @@ func (c *AzureConnection) ConnectionString() string {
 	)
 }
 
+/** Verify that the connection has enough access to read a VDS from Blob Store
+ *
+ * The minimum required access in a SAS token in order to read a VDS file from
+ * Blob Store is:
+ *
+ * sp  (Allow Premissions)      : 'r' (Read)
+ * srt (Allowed Resource Types) : 'c' and 'o' (Container and Object/blob)
+ *
+ * This function does a HEAD request to the VolumeDataLayout in the VDS, in
+ * form of a get properties request [1] in order to verify read access, and
+ * that the token is generally valid.
+ *
+ * The HEAD request itself is not sufficient tho, as it will not verify
+ * the container requirement in 'srt'. This is verifies manually.
+ *
+ * [1] https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob-properties
+ */
+func (c *AzureConnection) HasReadAccess() bool {
+	if err := validateSasSrt(c); err != nil {
+		return false
+	}
+
+	client, err := blob.NewClientWithNoCredential(
```

Comment:
What is our approach if someone has access to Volume.. file only? Like they got direct SAS to `https://someblob.blob.core.windows.net/testdata/well_known/well_known_default/VolumeDataLayout?sp=r&st=2023-01-02T16:01:35Z&se=2023-01-03T00:01:35Z&spr=https&sv=2021-06-08&sr=b&sig=stuff`
blob only (for example someone gave it to them so they can scrap some general info about our files. I can create one for Volume... file, able to read it, but can't read the Layout... file in the same vds).

Do I understand correctly that in that case we will return full data to the user, regardless them having wrong access? At least I seem able to do so.

Do we think that is fine and no big deal?

Reply:
Yes that is correct! We've talked about it and concluded that this requirement is good enough. But maybe I should document it better. I've documented it on the caching argument in the command line tool (main). Should I make the same note somewhere else too? What do you think?

Reply:
As you pointed out, we can be more strict than that. For Account SAS tokens I will verify that the 'srt' parameter contains either 'c' or 's' in addition to the 'b' that is verified by the HEAD request. This will guarantee that the sas token has access to all blob in the same directory and subdirectories as where the VolumeDataLayout resides.
For Service- and User delegation SAS tokens we verify that 'sr' contains 'c' or 'd' to the same effect